streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -646,18 +648,39 @@ public Subscription subscription(final Set<String> topics) {
                     for (final TaskId id : state.activeTasks()) {
                         topicPartitions.addAll(partitionsForTask.get(id));
                     }
-
                     partitionsByHostState.put(hostInfo, topicPartitions);
                 }
             }
+        } else if (minReceivedMetadataVersion >= 4) {
```

Comment:
could we reorder flow as below for better readability & progression:
if(version >= 4) { .... }
else if(version >=2) { .... }
..

Reply:
Will do it.